fuzzer/src/main.rs

```diff
@@ -90,23 +223,91 @@ fn fuzz_one_input(data: &[u8]){
 		Option<Vec<u8>>,
 		Vec<u32>,
 		LinkedList<u8>,
-		BTreeMap<String, Vec<u8>>,
-		BTreeMap<u8, u8>,
-		BTreeSet<u32>,
 		VecDeque<u8>,
-		BinaryHeap<u32>,
 		MockStruct,
 		MockEnum,
 		BitVec<Msb0, u8>,
 		BitVec<Msb0, u32>,
 		Duration,
+	};
+	// Types for which we wish to apply the "sorted" method.
+	fuzz_decoder! {
+		round_trip_sorted;
+		data;
+		BinaryHeapWrapper,
+	};
+	// Types which we only wish to decode.
+	fuzz_decoder! {
+		only_decode;
+		data;
+		BTreeMap<String, Vec<u8>>,
+		BTreeMap<u8, u8>,
+		BTreeSet<u32>,
+	};
+}
+
+macro_rules! fuzz_encoder {
+	() => {};
+	($( $type:ty, )*) => {
+		$(fuzz!(|data: $type| { fuzz_encode(data) });)*
+	};
+}
+
+fn fuzz_encode<T: Encode + Decode + Clone + PartialEq + std::fmt::Debug> (data: T) {
+	let original = data.clone();
+	let mut obj: &[u8] = &data.encode();
+	let decoded = <T>::decode(&mut obj);
+	if let Ok(object) = decoded {
```

Comment:
Again, please use a match.

Reply:
Argh haha yes I committed the change yesterday but forgot to push! :)